usecases/objects/auto_schema.go

```diff
@@ -492,3 +493,81 @@ func (m *autoSchemaManager) determineNestedPropertiesOfArray(valArray []interfac
 
 	return nestedProperties, nil
 }
+
+type classTenants map[*models.Class]map[string]struct{}
+
+func (m *autoSchemaManager) autoTenants(ctx context.Context,
+	principal *models.Principal, objects []*models.Object,
+) error {
+	var (
+		ct         = make(classTenants)
+		autoTenant = false
+	)
+
+	for _, obj := range objects {
+		// TODO: track the classes we have already checked
+		//       before calling getClass again
+		o := models.Object{Class: obj.Class}
+		class, err := m.getClass(principal, &o)
+		if err != nil {
+			return err
+		}
+
+		// class not found in schema. this will have already
+		// been caught by b.validateObjectsConcurrently, and
+		// will be handled accordingly, so we don't need to
+		// do anything here
+		if class == nil {
+			continue
+		}
+
+		if autoTenant || shouldAutoCreateTenants(class) {
+			if !autoTenant {
+				autoTenant = true
+			}
+			if _, ok := ct[class]; !ok {
+				ct[class] = map[string]struct{}{obj.Tenant: {}}
+				continue
+			}
+			ct[class][obj.Tenant] = struct{}{}
+		}
+	}
+
+	if autoTenant {
```

Comment:
`if` can be skipped. no tenant will be added if `ct` map is empty

Reply:
Agreed, removed